dwave_networkx/algorithms/partitioning.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2021 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from dwave_networkx import chimera_graph
+from dimod import DiscreteQuadraticModel
+from dwave.system import LeapHybridDQMSampler
+import networkx as nx
+import numpy as np
+
+__all__ = ["partition", "weighted_partition"]
+
+
+def partition(G, num_partitions=2, lagrange=4, sampler=None, **sampler_args):
+    """Returns an approximate k-partition of G.
+
+    Defines an Ising problem with ground states corresponding to a
+    balanced k-partition of G and uses the sampler to sample from it.
+
+    A k-partition is a collection of k subsets of the vertices
+    of G such that each vertex is in exactly one subset, and
+    the sum of the number of edges between vertices in different
+    subsets is as small as possible.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        The graph on which to find a partition.
+
+    num_partitions : int, optional (default 2)
+        The number of subsets in the desired partition.
+
+    lagrange: float, optional (default 4)
+        The value of the lagrange parameter used to enforce
+        the constraint that each partition is of equal size.
+
+    sampler
+        A discrete quadratic model sampler. A sampler is a process that
+        samples from low energy states in models defined by an Ising
+        equation or a Quadratic Unconstrained Binary Optimization
+        Problem (QUBO). A sampler is expected to have a 'sample_dqm'
+        method. A sampler is expected to return an iterable of samples,
+        in order of increasing energy. If no sampler is provided, one
+        must be provided using the `set_default_sampler` function.
+
+    sampler_args
+        Additional keyword parameters are passed to the sampler.
+
+    Returns
+    -------
+    node_partition : dict
+        The partition as a dictionary mapping each node to subsets labelled
+        as integers {0, 1, 2, ... num_partitions}.
```

Comment:
The dict shown is just the keys

Reply:
This was just supposed to be the set of subset labels - integers up to num_partitions. I could replace the curly brackets or just use words "labelled as integers from 0 to num_partitions." Would it be clearer to include a concrete example of integer labelled nodes a la {0: 0, 1: 0, 2: 1, ... }?